src/Core/src/Platform/Windows/KeyboardAcceleratorExtensions.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Input;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Windows.System;
+
+namespace Microsoft.Maui.Platform
+{
+	public static class KeyboardAcceleratorExtensions
+	{
+		public static void UpdateAccelerator(this MenuFlyoutItemBase platformView, IMenuFlyoutItem menuFlyoutItem)
+		{
+			var keyboardAccelerators = menuFlyoutItem.Accelerator.ToPlatform();
+
+			if (keyboardAccelerators is null)
+				return;
+
+			foreach (var keyboardAccelerator in keyboardAccelerators)
+			{
+				platformView.KeyboardAccelerators.Add(keyboardAccelerator);
+			}
+		}
+		
+		public static IList<KeyboardAccelerator>? ToPlatform(this IAccelerator accelerator)
+		{
+			if (accelerator is null)
+				return null;
+
+			List<KeyboardAccelerator> result = new List<KeyboardAccelerator>();
+
+			var keys = accelerator.Keys;
```

Comment:
Looks like you are iterating over all the keys and accessing the via the index, is that correct?  Because Keys are an IEnumerable, the order of the keys is not guaranteed, so you might get bugs due to the definition of you interface. 

The ordering of the items in an IEnumerable depends the implementation, if the user/implemenator of the IAccelerator uses a List, the order would be the same as the one added, but if it uses a HashSet, that order is not guaranteed, and both implemente the IEnumerable interface. 

If you want to guarantee the map, you should use a map.

Reply:
Good point, almost all the cases will use https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Accelerator.cs#L31 with List and no issues, but it would be possible to create something more custom using a Hashset.

Reply:
the issue is that you are using a IEnumerable, so there is no guarantee, meaning that you API design is very very error prone. Also, the size of the two enumerable is not guaranteed to be the same. 

The comment is more related to several design decisions:

1. You are using a Interface, IAccelerator that contains two interfaces that do not guarantee nor size equality nor order equality.
2. You have at least 2 bugs due to the design:
  a. You do not make sure that the size is the same.
  b. You do not make sure that the sorting is the same. ( And not, toArray won't help here
3. You have several performance hits with Count, used in every single iteration.

I would rethink the API to move to a situation in which the order of the IEnumerable is not needed. The length of the two things are always the same. For example, and IEnumerable of a tuple would guarantee that this method does not have the two bugs I mentioned. Then moving count out of the loop would also help.

Reply:
Are they meant to be the same number? What about Ctrl+Alt+X? That is 2 modifiers and 1 letter.

And with order, maybe this should just be an IList?

Reply:
This should be a list of tuples that match the modifiers + the letter, something a la IEnumerable<(string []Modifiers, string key)> and work with them correctly.

Reply:
Do we support the multi-letter way? Ctrl+K,D or Ctrl+K, Ctrl+D - the way VS does it. Not sure if this is an OS or VS feature. But yeah, Tuples are the way.

Reply:
@jsuarezruiz to clarify, you mean multiple modifiers, right? From our chat earlier, I think we established that an accelerator can have multiple modifiers but must have one and only one key.

based on this, I agree with what @PureWeen proposed offline that we can change the interface to just be string Key rather than an Enum.

I have the code locally and can push it up if you agree we're on the same page?

Reply:
Right, multiple modifiers.